controllers/link.js

```diff
@@ -1,28 +1,46 @@
 const express = require('express');
+
 const router = express.Router();
 const api = require('../api');
 
-router.post('/', function (req, res, next) {
-    // check first if target already exists (preventing db to be wasted)
-    let target = `${(req.headers.origin || process.env.HOST)}/${req.body.target}`;
-    api(req).get("/link/", { qs: { target: target } }).then(result => {
-        let existingLink = result.data[0];
-        if (!existingLink) {
-            api(req).post("/link/", { json: { target: target } }).then(data => {
-                data.newUrl = `${(req.headers.origin || process.env.HOST)}/link/${data._id}`;
-                res.json(data);
-            });
-        } else {
-            existingLink.newUrl = `${(req.headers.origin || process.env.HOST)}/link/${existingLink._id}`;
-            res.json(existingLink);
-        }
-    }).catch(err => next(err));
+const { HOST } = process.env;
+
+router.post('/', async (req, res, next) => {
+	// check first if target already exists (preventing db to be wasted)
+	const target = `${(req.headers.origin || HOST)}/${req.body.target}`;
+	try {
+		const result = await api(req).get('/link/', { qs: { target } });
+		const existingLink = result.data[0];
+		if (!existingLink) {
+			const data = await api(req).post('/link/', { json: { target } })
+			data.newUrl = `${(req.headers.origin || HOST)}/link/${data._id}`;
+			res.json(data);
+		} else {
+			existingLink.newUrl = `${(req.headers.origin || HOST)}/link/${existingLink._id}`;
+			res.json(existingLink);
+		}
```

Comment:
This logic should be moved to the server. The problem I discovered is, that the `generateRegistrationLink` Method in the administration controller also posts to the server to create shortlinks and because the server does not check for existing entries we have a problem.

You can reproduce this by going to:
`/administration/students/[studentID]/edit` and clicking on `Persönlichen Einladungslink generieren`. Everytime you open the modal a new database entry will be created and you will get a new shortlink.

This implementation also doesn't fullfill our acceptance criteria that all the logic should be implemented in the server.